metrics/config.go

```diff
@@ -20,40 +20,44 @@ import (
 	"errors"
 	"fmt"
 	"github.com/michaelquigley/pfxlog"
+	"time"
 )
 
 type Config struct {
-	handlers map[Handler]Handler
+	handlers       map[Handler]Handler
+	Source         string
+	Tags           map[string]string
+	ReportInterval time.Duration
```

Comment:
Wouldn't it be better to let each of the handlers decide how frequently they want to report their metrics? I would imagine we'd want the handler to just handle the metrics from our infrastructure, and then decide on whatever period they want, do something with the accumulated metrics.

Reply:
Yes and no. I started down that path, but there's some conceptual problems, though in practice, maybe no actual problems. 

So each Registry collects usage data. Each time we create a new metrics event, we flush all the usage data into it. So if you had multiple handlers with different schedules, figured out how to feed them usage data would get very complicated. 

On the other hand, in the controller where we actually have multiple reporters we're not collecting any usage data yet (that only happens in the routers). 

But on the third hand, metrics event handlers are going to get events from multiple systems (controller + routers), so for them, an interval doesn't really make sense, since they'll be getting data whenever the routers want to send it. So we're really just deciding how often we want to snapshot the controller metrics and push events to any handlers.

So all that conceptual messiness  why I backed off the interval per handler. If you want to talk it through, I'm happy to do that, though. There's probably room in there to be able to customize things per hander.

Reply:
I'm sure there's some conceptual messiness that I'm not seeing, but I feel like the relationship between the handler and the registry should probably be split so that the registry just works like a "reflector"... pushing all of the events to the registered handlers, and the handlers each individually decide how to roll up, or report those metrics, and on what interval. That kind of a model gives system extenders the maximum flexibility to do what they want with those event streams.

I suspect we're probably in a situation where the registry and the infrastructure is doing some of that "rolling up", and that's why things are a little confusing conceptually.

Reply:
It really comes down that we have two kinds of data. Usage data, and everything else. Usage data is consumed when polled. Everything else, you just get the current snapshot. So for everything other than usage data, you can have multiple data sinks, polling at odd intervals without any problem. For the usage data, you need a single sink, or at least a single distributor. 

Longer term, I think we can organize it so that we can isolate the usage data, or even just ignore it. There are other changes I want to make as well. But I don't want to do it now, as there are other higher priority items to tackle. Since this PR doesn't introduce the current paradigm, can we leave resolving this to the next round of metrics improvements?

Reply:
Sure. Getting onto a consolidated general approach of using handlers to consume metrics and usage data is probably a fine intermediate step for now.

Reply:
I didn't fully resolve this, but I did break out the usage components into an extension of the base registry functionality. Now only the UsageRegistry reports on an interval. The controller will use a regular Registry. For now the fabric is still pushing metrics on an interval. Down the line we can have Handlers do Registry.Poll on their own schedule. 

I'll wait a bit before I merge this up in case you want to take another look (also need to wait on the other foundation PR to get merged first anyway)